utils/conversions.ts

```diff
@@ -124,3 +133,56 @@ export function userContractToApiUser(
     type: user.type
   }).mapLeft(errors => new Error(errorsToReadableMessages(errors).join(" / ")));
 }
+
+export function groupContractToApiGroup(
+  group: GroupContract
+): Either<Error, ApiGroup> {
+  return Group.decode(
+    removeNullProperties({
+      built_in: group.builtIn,
+      description: group.description,
+      display_name: group.displayName,
+      external_id: group.externalId,
+      group_contract_type: group.groupContractType,
+      id: group.id,
+      name: group.name,
+      type: group.type
+    })
+  ).mapLeft(errors => new Error(errorsToReadableMessages(errors).join(" / ")));
+}
+
+export function subscriptionContractToApiSubscription(
+  subscription: SubscriptionContract
+): Either<Error, ApiSubscription> {
+  return Subscription.decode(
+    removeNullProperties({
+      allow_tracing: subscription.allowTracing,
+      created_date: subscription.createdDate,
+      display_name: subscription.displayName,
+      end_date: subscription.endDate,
+      expiration_date: subscription.expirationDate,
+      id: subscription.id,
+      name: subscription.name,
+      notification_date: subscription.notificationDate,
+      owner_id: subscription.ownerId,
+      primary_key: subscription.primaryKey,
+      scope: subscription.scope,
+      secondary_key: subscription.secondaryKey,
+      start_date: subscription.startDate,
+      state: subscription.state,
+      state_comment: subscription.stateComment,
+      type: subscription.type
+    })
+  ).mapLeft(errors => new Error(errorsToReadableMessages(errors).join(" / ")));
+}
+
+function removeNullProperties<T>(obj: T): unknown {
```

Comment:
why do we need this ? can we just return the object with eventually null values ?

Reply:
The object returned here will be used to create a response payload. The type and codec of the payload are generated from the specs file, which is written in swagger format, and this format does not support the `null` value.